### ISSUE TITLE
False positive for SC2004 - $ unnecessary in (( )) when using associative array in calculation

### ISSUE THREAD

**Issue by brother** [koalaman/shellcheck#183]
```
brother /tmp$ cat test.sh 
#!/bin/bash

declare -A kaka

banan=foobar

kaka[$banan]=0

# Doesn't work but doesn't trigger any shellcheck warnings
(( kaka[banan] += 42 ))
echo ${kaka[$banan]}

# Works but shellcheck SC2004 triggers
(( kaka[$banan] += 42 ))
echo ${kaka[$banan]}
brother /tmp$ ./test.sh 
0
42
brother /tmp$ shellcheck test.sh 

In test.sh line 14:
(( kaka[$banan] += 42 ))
        ^-- SC2004: $ on variables in (( )) is unnecessary.

brother /tmp$ 
```

Using latest git HEAD aka 64cc7c691.
